### ISSUE TITLE
Xcode 15.1 and visionOS support

### ISSUE THREAD

**Issue by paulb777** [firebase/firebase-ios-sdk#11890]
### Description

From the Xcode 15.1 beta release notes:

- Fixed: When building with the visionOS SDK, TargetConditionals.h incorrectly sets TARGET_OS_IOS to 1 instead of 0. (112716373)

### Reproducing the issue

We need to evaluate all places in the code where `TARGET_OS_IOS` and determine the right path for visionOS.

### Firebase SDK Version

10.16

### Xcode Version

15.1

### Installation Method

N/A

### Firebase Product(s)

All

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by andrewheard** [comment]
Resolved in upcoming 10.17.0 release. See #11927, #11930, #11935, #11937, #11950 and #11938 for fixes.

**Comment by triplef** [comment]
Thanks for the feedback @paulb777. We now migrated to SPM which works great with visionOS.
